cmd/migrate_cmd.go

```diff
@@ -31,76 +41,127 @@ func migrate(cmd *cobra.Command, args []string) {
 		globalConfig.DB.Driver = u.Scheme
 	}
 
-	log := logrus.New()
-
-	pop.Debug = false
-	if globalConfig.Logging.Level != "" {
-		level, err := logrus.ParseLevel(globalConfig.Logging.Level)
-		if err != nil {
-			log.Fatalf("Failed to parse log level: %+v", err)
-		}
-		log.SetLevel(level)
-		if level == logrus.DebugLevel {
-			// Set to true to display query info
-			pop.Debug = true
-		}
-		if level != logrus.DebugLevel {
-			var noopLogger = func(lvl logging.Level, s string, args ...interface{}) {
-				return
-			}
-			// Hide pop migration logging
-			pop.SetLogger(noopLogger)
-		}
+	source := &sqlmigrate.FileMigrationSource{
+		Dir: "migrations/",
 	}
 
-	deets := &pop.ConnectionDetails{
-		Dialect: globalConfig.DB.Driver,
-		URL:     globalConfig.DB.URL,
-	}
-	deets.Options = map[string]string{
-		"migration_table_name": "schema_migrations",
+	db, err := sql.Open("pgx", globalConfig.DB.URL)
+	if err != nil {
+		logrus.Fatalf("Failed to connect to the database: %v", err.Error())
 	}
+	defer db.Close()
 
-	db, err := pop.NewConnection(deets)
+	// create migrations table for sql-migrate
+	_, err = db.Exec(`create table if not exists migrations (
+		id text, 
+		applied_at timestamptz, 
+		primary key (id)
+	)`)
 	if err != nil {
-		log.Fatalf("%+v", errors.Wrap(err, "opening db connection"))
+		logrus.Fatalln(err)
 	}
-	defer db.Close()
 
-	if err := db.Open(); err != nil {
-		log.Fatalf("%+v", errors.Wrap(err, "checking database connection"))
+	// check if schema_migrations table exists
+	res, err := db.Query("select exists(select from pg_tables where schemaname = 'auth' and tablename = 'schema_migrations')")
+	if err != nil {
+		logrus.Fatalln(err)
+	}
+	defer res.Close()
+	var exists bool
+	for res.Next() {
+		if err := res.Scan(&exists); err != nil {
+			logrus.Fatalln(err)
+		}
 	}
 
-	log.Debugf("Reading migrations from %s", globalConfig.DB.MigrationsPath)
-	mig, err := pop.NewFileMigrator(globalConfig.DB.MigrationsPath, db)
+	migrationFiles, err := ioutil.ReadDir("./migrations/")
 	if err != nil {
-		log.Fatalf("%+v", errors.Wrap(err, "creating db migrator"))
+		logrus.Fatalln(err)
 	}
-	log.Debugf("before status")
 
-	if log.Level == logrus.DebugLevel {
-		err = mig.Status(os.Stdout)
+	if exists {
+		// get versions from schema_migrations
+		res, err = db.Query("select version from schema_migrations")
 		if err != nil {
-			log.Fatalf("%+v", errors.Wrap(err, "migration status"))
+			logrus.Fatalln(err)
+		}
+		defer res.Close()
+		for res.Next() {
```

Comment:
Would it be worthwhile to check for errors at end of block like in:  http://go-database-sql.org/retrieving.html#fetching-data-from-the-database ?

```go
err = rows.Err()
if err != nil {
	log.Fatal(err)
}
```